src/backend/opencl/program.cpp

```diff
@@ -70,4 +79,38 @@ void buildProgram(cl::Program &prog, const int num_files, const char **ker_strs,
         throw;
     }
 }
+
+cl::Program buildProgram(const std::vector<std::string>& kernelSources,
+                         const std::vector<std::string>& compileOpts) {
+    cl::Program retVal;
+    try {
+        static const std::string defaults =
+            std::string(" -D dim_t=") +
+            std::string(dtype_traits<dim_t>::getName());
+
+        auto device = getDevice();
+
+        const std::string cl_std =
+            std::string(" -cl-std=CL") +
+            device.getInfo<CL_DEVICE_OPENCL_C_VERSION>().substr(9, 3);
+
+        Program::Sources sources;
+        sources.emplace_back(DEFAULT_MACROS_STR);
+        sources.emplace_back(KParam_hpp, KParam_hpp_len);
+
+        for (auto ksrc : kernelSources) { sources.emplace_back(ksrc); }
+
+        retVal = cl::Program(getContext(), sources);
+
+        ostringstream options;
+        for (auto &opt: compileOpts) { options << opt; }
+
+        retVal.build({device}, (cl_std + defaults + options.str()).c_str());
+    } catch (...) {
```

Comment:
I merely copied the implementation from other `buildProgram` version. I will look into why it is `(...)` instead of `cl::Error`.

I don't this should be in `common`, it has OpenCL specific code.

Reply:
We already have a catch for cl::Error in that file. I wonder why the ... was added.